specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-05-01-preview/Migrations.json

```diff
@@ -0,0 +1,826 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{targetDBServerName}/checkMigrationNameAvailability": {
+      "post": {
+        "summary": "Check migration name validity and availability",
+        "description": "This method checks whether a proposed migration name is valid and available.",
+        "tags": [
+          "Custom operation",
+          "POST"
+        ],
+        "operationId": "CheckMigrationNameAvailability",
+        "x-ms-examples": {
+          "CheckMigrationNameAvailability": {
+            "$ref": "./examples/CheckMigrationNameAvailability.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetDBServerSubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetDBServerResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetDBServerNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MigrationNameAvailabilityResource"
+            },
+            "description": "The required parameters for checking if a migration name is available."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MigrationNameAvailabilityResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "MigrationResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/MigrationResourceProperties",
+          "description": "Migration resource properties.",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/systemData",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Represents a migration resource."
+    },
+    "MigrationResourceProperties": {
+      "type": "object",
+      "properties": {
+        "migrationId": {
+          "type": "string",
+          "readOnly": true
+        },
+        "migrationName": {
```

Comment:
>migrationName [](http://example.com/codeflow?start=9&length=13)

what is the difference between this and the name of the resource that the user supplied on the PUT?

Reply:
@pilor migrationName and nameOfMigration both are same 

when kept as migrationName everywhere got this error in lint errors

<html>
<body>
<!--StartFragment-->

❌ RepeatedPathInfo | The 'migrationName' already appears in the path, please don't repeat it in the request body.Location: Microsoft.DBforPostgreSQL/preview/2022-05-01-preview/Migrations.json#L62 | RPC-Put-V1-05
-- | -- | --


<!--EndFragment-->
</body>
</html>❌ [RepeatedPathInfo](https://github.com/Azure/azure-openapi-validator/blob/main/docs/repeated-path-info.md)	The 'migrationName' already appears in the path, please don't repeat it in the request body.
Location: [Microsoft.DBforPostgreSQL/preview/2022-05-01-preview/Migrations.json#L62](https://github.com/Azure/azure-rest-api-specs/blob/86a7271a72b43a6da4991458eae30fb1c1e21b6d/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-05-01-preview/Migrations.json#L62)	RPC-Put-V1-05

So made it as nameOfMigration. But both are same

Reply:
What that rule is trying to say is that the name of the resource should go in the "name" property and not anything else. Whatever the user supplied in the URI as the name must become the name of the resource

Reply:
So should I change it from nameOfMigration to name in resource? @pilor

Reply:
Can I discuss about this in personal? Didn't understand fully. @pilor

Reply:
@pilor I removed migrationName property from MigrationResourceProperties in Migration.json, and corresponding examples.
With this I see this issue got resolved, is this sufficient or else do we need to fix it in backend RP as well?
Even if backend responds with migrationName value in response will that be a concern as we removed it in swagger.

Reply:
Swagger needs to map exactly with the actual REST API request/response

Reply:
@pilor will file a bug to remove this from backend and that fix will be done in next release ok backend
As this is for public preview only we will fix this for upcoming GA. Is this fine and can we merge this PR?

Reply:
@xiaoxuqi-ms is the SDK reviewer who will handle completion of the review and merging